### PR TITLE
Remove ClazzRef - use TypingResult/TypedClass instead

### PR DESCRIPTION
**Pull request by mproch**
1. Let's remove ClazzRef class. It's role can be served by TypingResult, specializing to TypedClass in some places
2. We have to be careful with creating TypeClass object - as TypedClass[Any] should be Unknown. Let's not invoke TypedClass constructor directly, but with factory methods which in most cases return safe TypingResult. When we **really** need TypedClass we have to make sure it's not Unknown
3. Let's leave REST API backward compatible ATM. I've intentionally serialize Unknown in TypeEncoders just like before. We may change that in the future, but let's see how this change works out before
4. realReturnType is simplified to Class[_]

### CODE REVIEW

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/WithExplicitMethodToInvoke.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ trait WithExplicitMethodToInvoke {
 
   def returnType: TypingResult
 
-  def realReturnType: TypingResult
+  def realReturnType: Class[_]
```
Comment (arkadius) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/WithExplicitMethodToInvoke.scala]: Please also rename all `realReturnType` fields to `runtimeClass`. Just like in `Parameter` class.

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ServiceInvoker.scala [hunk 2]
```diff
@@ -65,11 +65,7 @@ object ServiceInvoker {
   )
 
   def apply(objectWithMethodDef: ObjectWithMethodDef, collector: Option[ServiceInvocationCollector] = None): ServiceInvoker = {
-    val detectedReturnType = (objectWithMethodDef.methodDef.realReturnType match {
-      case tc: SingleTypingResult =>
-        Some(tc.objType.klass)
-      case _ => None
-    }).getOrElse(classOf[Any])
+    val detectedReturnType = objectWithMethodDef.methodDef.realReturnType
```
Comment (arkadius) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ServiceInvoker.scala]: `detectedRuntimeClass`

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala [hunk 3]
```diff
@@ -185,7 +184,7 @@ object node {
 
     object SubprocessClazzRef {
 
-      def apply[T: TypeTag]: SubprocessClazzRef = SubprocessClazzRef(ClazzRef.fromDetailedType[T].refClazzName)
+      def apply[T: ClassTag]: SubprocessClazzRef = SubprocessClazzRef(implicitly[ClassTag[T]].runtimeClass.getName)
```
Comment (arkadius) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala]: WDYT about leaving comment on `SubprocessClazzRef` level that we should also replace this class with `TypingResult`. I know that for now we don't have options on UI to declare that input will be `TypedObjectTypingResult` or `TypedUnion` but can't see any blocker to have such option on BE side. Thanks to that we will have one class less and will be cleaner (we won't mix up compilation (TypingResult) with runtime (classes)).

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/uiconfig/defaults/ParameterEvaluatorSpec.scala [hunk 4]
```diff
@@ -3,34 +3,32 @@ package pl.touk.nussknacker.ui.process.uiconfig.defaults
...
   private val DEFAULT_NODE_NAME = "undefined"
   private val DEFAULT_PARAMETER_VALUE = "defVal"
   private val DEFINED_NODE_NAME = "defined"
 
-  private def dummyParam(nodeName: String,
-                         paramName: String,
-                         classRef: ClazzRef) =
-    NodeDefinition(nodeName, List(Parameter(paramName, classRef)))
+  private def dummyParam[T:ClassTag](nodeName: String,
+                         paramName: String) =
```
Comment (arkadius) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/process/uiconfig/defaults/ParameterEvaluatorSpec.scala]: can you fix formatting here and below?

Comment (mproch) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/process/uiconfig/defaults/ParameterEvaluatorSpec.scala]: :+1: , done

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala (hunk 3 above)
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala]: In fact we already have this TODO few lines above :D I added clarification

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ServiceInvoker.scala (hunk 2 above)
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ServiceInvoker.scala]: :+1: , done

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/WithExplicitMethodToInvoke.scala (hunk 1 above)
Comment (mproch) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/WithExplicitMethodToInvoke.scala]: :+1: , done

**Pull request merged**